tests/test_code_replacement.py

```diff
@@ -1621,3 +1662,80 @@ def new_function2(value):
         project_root_path=Path(__file__).resolve().parent.resolve(),
     )
     assert new_code == original_code
+
+def test_global_reassignment() -> None:
+    original_code = """a=1
+print("Hello world")
+def some_fn():
+    print("did noting")
+class NewClass:
+    def __init__(self, name):
+        self.name = name
+    def __call__(self, value):
+        return "I am still old"
+    def new_function2(value):
+        return cst.ensure_type(value, str)
+    """
+    optimized_code = """import numpy as np
+def some_fn():
+    a=np.zeros(10)
+    print("did something")
+class NewClass:
+    def __init__(self, name):
+        self.name = name
+    def __call__(self, value):
+        return "I am still old"
+    def new_function2(value):
+        return cst.ensure_type(value, str)
+a=2
+print("Hello world")
+    """
+    expected_code = """import numpy as np
+print("Hello world")
+
+a=2
```

Comment:
can you write more tests for this?
in the original code global before imports, global after function defintions, global conditionally defined behind if conditions. 
you can think of more